collector/lib/Diagnostics.h

```diff
@@ -15,29 +15,54 @@ class IDiagnostics {
 
 class StartupDiagnostics : public IDiagnostics {
  public:
-  StartupDiagnostics()
-      : connectedToSensor_(false),
-        kernelDriverDownloaded_(false),
-        kernelDriverLoaded_(false) {}
-
   void Log() {
     CLOG(INFO) << "";
     CLOG(INFO) << "== Collector Startup Diagnostics: ==";
     CLOG(INFO) << " Connected to Sensor?       " << std::boolalpha << connectedToSensor_;
-    CLOG(INFO) << " Kernel driver available?   " << std::boolalpha << kernelDriverDownloaded_;
-    CLOG(INFO) << " Driver loaded into kernel? " << std::boolalpha << kernelDriverLoaded_;
+
+    CLOG(INFO) << " Kernel driver candidates:";
+    for (const auto& line : driverDiagnostics_) {
+      CLOG(INFO) << line;
+    }
+
     CLOG(INFO) << "====================================";
     CLOG(INFO) << "";
+
+    driverDiagnostics_.clear();
+  }
+
+  static StartupDiagnostics& GetInstance() {
+    static StartupDiagnostics instance;
+
+    return instance;
   }
 
   void ConnectedToSensor() { connectedToSensor_ = true; }
-  void KernelDriverDownloaded() { kernelDriverDownloaded_ = true; }
-  void KernelDriverLoaded() { kernelDriverLoaded_ = true; }
+
+  void DriverUnavailable(const std::string& candidate) {
```

Comment:
Well, unavailable means unavailable, as in, it's nowhere to be found (in the image, online, etc..). Adding the distinction between local or downloadable can be tricky, we would need to add some logic to the `GetKernelObject` function and I don't see it as necessarily useful.

The search path is already print in previous logs iirc and, excluding the candidate from `COLLECTOR_DRIVER`, it will always be `/kernel-modules/<candidate name>`.

Reply:
What I mean here is that we operate with an object "candidate", but diagnostics get only the name. Wouldn't it be beneficial to provide the full candidate there? There is no need to do any distinction, just dump all the info we got into the output.

Reply:
My understanding is that the message output by this diagnostics is a meant to be a short, meaningful message that users/support should be able to quickly interpret, if we start dumping things in it, it will lose this.

@Stringy, you put together the feature, what do you think?